src/PIL/PpmImagePlugin.py

```diff
@@ -49,26 +49,46 @@ class PpmImageFile(ImageFile.ImageFile):
     format = "PPM"
     format_description = "Pbmplus image"
 
-    def _token(self, s=b""):
-        while True:  # read until next whitespace
+    def _read_magic(self):
+        magic = b""
+        # read until whitespace or limit of 6 characters
+        for _ in range(6):
             c = self.fp.read(1)
             if not c or c in b_whitespace:
                 break
-            if c > b"\x79":
-                raise ValueError("Expected ASCII value, found binary")
```

Comment:
This was removed because, as point 4 of #5104 states, b"\x79" is "y"... which is confusing, because that 's still ASCII. The intention was probably to the limit the start of the magic to "Py" and nothing alphabetically after that.